### ISSUE TITLE
Current status

### ISSUE THREAD

**Issue by alexandremottet** [manuels/unix-toolbox.js#1]
Hi,

I would like to know what is the current status of those project and in particular the "unix-toolbox.js-imagemagick" ? Is it active and working ?

Regards

**Comment by manuels** [comment]
Yes, imagemagick is working fine. Checkout the ./demo folder to see how to use it

**Comment by alexandremottet** [comment]
Ok, I've already tried it and several errors comes out :
![unixtoolbox-js-errors](https://f.cloud.github.com/assets/617692/1198545/da154b26-24f9-11e3-99c5-8618aa34db8b.png)

Does it came from my computer (chrome) ?

**Comment by manuels** [comment]
I found the bug: I didn't update the submodule to the latest version.
You can try again, it should work now.

**Comment by alexandremottet** [comment]
Ok, thanks for reply so quickly !

Works for jpg/png but I was unable to convert a G4 tif file image with unix-toolbox.
The same image goes ok in command line (convert "g4.tif" "g4.jpg").

![unixtoolbox-js-errors](https://f.cloud.github.com/assets/617692/1199219/f803d580-2508-11e3-8ad2-7ed6ebafdbd1.png)

The test file : http://www.fileformat.info/format/tiff/sample/ff5798fb1a324c099af642690d161877/download

When you compile ImageMagick, did you add some exclusion ?

**Comment by manuels** [comment]
Yes, I did not include all image format libraries. However, I included tiff. So it's strange that this does not work.
What's the mime type of your tif file? is G4 tif different than the regular tif format?

**Comment by manuels** [comment]
Oh, G4 tif means group 4 tiff (Raw CCITT Group4) according to `identify -list format`.

Probably I did not include this format library...

**Comment by alexandremottet** [comment]
Ok. The mime-type is "image/tiff" given "file --mime-type g4.tif" command line.
This is a regular type for register tiff file (fax).

**Comment by manuels** [comment]
Can you run `convert -identify g4.tif` (on your computer and in the browser) and post the output?

**Comment by alexandremottet** [comment]
It give me this line :
g4.tif TIFF 2464x3248 2464x3248+0+0 1-bit Bilevel Gray 57.3KB 0.050u 0:00.049

**Comment by manuels** [comment]
^^Is that the output for both or did you just try it in the console?

**Comment by manuels** [comment]
Can you publish this image anywhere so I can try it myself?

**Comment by alexandremottet** [comment]
I used this image example : http://www.fileformat.info/format/tiff/sample/ff5798fb1a324c099af642690d161877/download

Now, i'll just try in the console.

**Comment by manuels** [comment]
I'm not sure if the file is valid. Even my native imagemagick (debian 6.7.7-10 2013-09-01 Q16) tells me
`convert.im6: no images defined 'g4.tif' @ error/convert.c/ConvertImageCommand/3044.`

**Comment by alexandremottet** [comment]
I think this image is valid (from file format.info). This command line (convert -identify "g4.tif" "g4.jpg") work on my computer (mac osx).

What is yout ImageMagick version ?

**Comment by manuels** [comment]
It's `6.7.7-10 2013-09-01 Q16` and yours?

**Comment by alexandremottet** [comment]
It's 6.8.0-7 2013-09-19.
I don't know if there is so much difference between those version.

**Comment by manuels** [comment]
Got it: I did not use the additional 'g4.jpg' parameter. So it also works with my native version.
But I did not figure out which delegate is responsible for GROUP4 images.

**Comment by alexandremottet** [comment]
The js version of "convert -identify .." gives :
<code>Magick: no decode delegate for this image format `/g4.tif' @ error/constitute.c/ReadImage/550.</code>

**Comment by manuels** [comment]
I think I got it: probably CCITT support is not included in libtiff. Right now I don't have my laptop (I lost the power supply and I'm waiting for the new one). As soon as I can use my laptop again, I can check whether this really is the issue and fix it.

**Comment by manuels** [comment]
Hmm, I don't get it!
[Line 8 in tif_config.h](https://github.com/manuels/unix-toolbox.js-libtiff/blob/master/src/libtiff/tif_config.h#L8) clearly states that CCITT Group 3 and 4 support should be included!
